### DISCUSSION TITLE
Empty functions suggested

### DISCUSSION THREAD

**Discussion by bannsec** [community/community#6631] [Copilot Conversations] [answered]
In playing with this, I've discovered many unfortunate failings. One such thing is when you get a function back that literally does nothing, no matter how hard you try to describe to the AI what you're interested in.

Example -- I'd like to get a c++ function to base64 encode a string. No matter what I do, i get this:
```c++
char *encode_base64(char *input) {
    // Takes in a string and returns a base64 encoded string
    return input;
}
```

Not sure if that non-function is better or worse than when i try the same thing in C, which returns a function that simply doesn't correctly b64 encode.

I actually get that "return input" or "return <specific value>" a LOT. I have other issues with Windows APIs not being used correctly, but that will be for a different thread.

**Comment by bannsec** [comment]
Another example I stumbled into is anything to do with hashes (md5/sha/etc)

```c++
// Creates sha256 hash of input string
char *sha256(char *input) {
    char *output = new char[65];
    return output;
}
```

**Reply by Urethramancer** [reply]
Crypto and hashes are the things I'm seeing the worst results for in other languages too. I think the AI has been trained to fear i.t

**Comment by dvd-22** [comment, marked as answer]
When you press <kbd>CTRL + Enter</kbd> it shows uo 10 suggestions. Are all suggestions empty?

**Reply by bannsec** [reply]
Ah. Ok. I wasn't aware of the multiple suggestions. Still seems not useful to have empty suggestions but better ones did exist.

**Reply by Urethramancer** [reply]
Yeah, a filter to skip empty functions as an option at all would be better. Any scaffolding tools I use already makes stub code anyway.
